File: fixtures/scale/swift/large/Families/BlockReverse/BlockReverseDescriptor.swift
struct BlockReverseDescriptor: FamilyDescriptor {
    var family: String { "blockreverse" }
    var suite: String { "block" }
    var kind: CipherKind { .block }

    func cipher() -> Cipher {
        return BlockReverseCipher()
    }

    func vectors() -> [TestVector] {
        return BlockReverseVectors.all()
    }
}

File: fixtures/scale/csharp/large/Families/BlockReverse/BlockReverseDescriptor.cs
using System.Collections.Generic;
using CipherLab.Abstractions;

namespace CipherLab.Families.BlockReverse
{
    public class BlockReverseDescriptor : IFamilyDescriptor
    {
        public string Family => "blockreverse";
        public string Suite => "block";
        public CipherKind Kind => CipherKind.Block;

        public ICipher Cipher()
        {
            return new BlockReverseCipher();
        }

        public List<TestVector> Vectors()
        {
            return BlockReverseVectors.All();
        }
    }
}
